### DISCUSSION TITLE
Redo migration that is not the latest

### DISCUSSION THREAD

**Discussion by myz-dev** [diesel-rs/diesel#2843] [Q&A] [answered]
Hello everyone,

I am updating the structure of my tables so I need to redo their migrations.
With the diesel CLI I am running `diesel migration redo` which only applies to the newest migration.
How do I select any other migration as the object of my command?
`diesel migration list` outputs:
```
Migrations:
  [X] 00000000000000_diesel_initial_setup
  [X] 2021-08-02-151557_create_rolls
  [X] 2021-08-03-064352_create_defect
  [X] 2021-08-03-065855_create_recipe
```

Can someone tell me how to form the command so that it targets any other migration?

**Comment by weiznich** [comment, marked as answer]
Migrations need to be applied in order, so there is just no way to redo a random migration without reverting all newer migrations first. 
That written: The current release only allows you to redo the last migration. If you want to redo an older one you need to revert all newer ones manually first. The next feature release will include some change that automates this process.

**Reply by myz-dev** [reply]
Hi,
thank you very much for your fast reply.
I understand, I could revert all migrations and run them again one after another. Thank you!
Edit for anyone else looking for an answer:
Just run `diesel migration revert` until all migrations that you want to redo are reverted. Then run `diesel migration run` which will run all pending migrations.
